### ISSUE TITLE
Questions Regarding Useage

### ISSUE THREAD

**Issue by andrewschluchter** [mwilliamson/stickytape#1]
Had some questions regarding stickytape; was directed by developer to upload as issues here.

There does not seem to be documentation regarding which platforms/operating systems stickytape works for. Slashes in examples appear to indicate UNIX-based systems, but I'm unsure about use on Windows, or which UNIX-based system(s). Also unsure about how to "stickytape" my python code based on its version (currently working with variations of both versions 2 and 3, and need resulting code for both).

Documentation also is unfamiliar to me regarding command line examples, e.g., the bit " . > " in the command line example:
stickytape scripts/blah --add-python-path . > /tmp/blah-standalone

The most successful attempt so far has produced a Python 3 script that contains mentions of all modules needed, but does not appear to include modules in the resultant script in any way.
I am happy to load my own code and attempt to upload exact errors as needed.

Thanks :)

**Comment by mwilliamson** [comment]
Stickytape was a fun little side-project, rather than anything I intended for anyone to use, so alternatives might suit you better. Having said that:

> There does not seem to be documentation regarding which platforms/operating systems stickytape works for. Slashes in examples appear to indicate UNIX-based systems, but I'm unsure about use on Windows, or which UNIX-based system(s).

I've only tried it on Linux. I'd expect it to also work on Windows since it also understands forward slashes as path separators.

> Also unsure about how to "stickytape" my python code based on its version (currently working with variations of both versions 2 and 3, and need resulting code for both).

Stickytape just writes out the code as originally written, so should support both Python 2 and 3.

> Documentation also is unfamiliar to me regarding command line examples, e.g., the bit " . > " in the command line example:
> stickytape scripts/blah --add-python-path . > /tmp/blah-standalone

The `>` is shell redirection. For instance:

```
echo 1 > x
```

will write `1` to the file `x`.

The `.` is the argument to `--add-python-path` i.e. it says to look for Python modules using the current directory as the root.

> The most successful attempt so far has produced a Python 3 script that contains mentions of all modules needed, but does not appear to include modules in the resultant script in any way.

What happens when you try running the script? Can you try producing the smallest example that still has issues and post that?

**Comment by andrewschluchter** [comment]
Thank you for the quick response. I will respond back with a small "stickytaped" example within 8 hours. Also, thank you for making this project. This was the first I had seen that would allow me to, hopefully, create a single-file, cross-platform Python script for batch-processing medical files. And side-project or not, some of our most critical scientific discoveries and projects throughout history were accidents and side-projects. 

TL;DR: I appreciate the work. I'll respond this evening.

**Comment by andrewschluchter** [comment]
After a couple of hours of testing on a couple of different platforms, I have the final output working, for the most part. Here are the results of what I've found, which may help to clarify things for others in the future:

While the path to the initial script is relative to the current working directory, the path of the output script is absolute. I assume this is likely standard protocol for UNIX systems, but I had not considered it among the options of what the issue could be.

Calling Python on Linux systems will default to Python 2, even if the shebang comment at the top of the script says otherwise (though this may simply be unique to certain platforms).

Lastly, it seems that the main issues in the code due to Python version 2 vs 3 essentially boiled down to issues in the "dicom" module itself. I will be investigating this module next.


Thank you very much for the help, and I'm sure I will be using this script again in the future.

**Comment by mwilliamson** [comment]
> While the path to the initial script is relative to the current working directory, the path of the output script is absolute. I assume this is likely standard protocol for UNIX systems, but I had not considered it among the options of what the issue could be.

You mean the final file that gets written? If you're using bash redirection, then that path can be relative. For instance, this writes out a file called `x` in the current directory:

```
echo 1 > x
```

> Calling Python on Linux systems will default to Python 2, even if the shebang comment at the top of the script says otherwise (though this may simply be unique to certain platforms).

The shebang is only used if the file is marked as executable (e.g. `chmod +x run.py`) and the script is executed directly i.e. without a Python interpreter. For instance:

```
python run.py
```

will always run `run.py` with whatever interpreter `python` refers to, regardless of the shebang in `run.py`. However, running the script directly:

```
./run.py
```

will run `run.py` with whatever is specified in the shebang.

**Comment by andrewschluchter** [comment]
Interesting. I'll continue to do testing on stickytape to become more familiar with it, and with Python operations on the terminal.

Thank you for all of the help.

**Comment by XxoraxX** [comment]
@andrewschluchter hi did stickytape  worked with you because i have problem to understand the script option

**Comment by andrewschluchter** [comment]
Yes, I was able to make it work.

If you have a specific question, I will be happy to try and answer it.
